### ISSUE TITLE
Support for x86 platforms

### ISSUE THREAD

**Issue by jaselee0905** [velopack/velopack#5]
I've been using clowd.squirrel and would like to migrate to velopack but I can't create an installer for 32 bit systems.
Will this come in the future or is 32bit support dropping completely

**Comment by caesay** [comment]
Can I ask out of curiosity what platform you're trying to support which needs x86?

**Comment by jaselee0905** [comment]
its for windows 7 we know its eol but we still have many users that are using it

**Comment by ysmoradi** [comment]
Greetings
Is [this](https://github.com/velopack/velopack/commit/286923d923cc15127b4f4f31d8ab53f936f8fb34) commit related to this issue?

**Comment by caesay** [comment]
Only in the most basic sense. I couldn't get x86 builds working locally on my windows machine. I mean, Rust would compile but the resulting assemblies were broken (looked like some kind of link failure). So I thought I would try it in CI via ubuntu but as you can see from the actions history, none of that worked - and I moved my fiddling to another branch.

**Comment by ysmoradi** [comment]
I asked this question from gpt:

I'm trying to cross compile rust code, but for the windows x86, I receive the following error: "cannot find -lpropsys"

And received this answer, hope it helps:

The error "cannot find -lpropsys" indicates that the linker is unable to find the library named `propsys.lib` during the cross-compilation process for Windows x86 (`i686-pc-windows-gnu`). To resolve this issue, you may need to take the following steps:

1. **Ensure Required Libraries are Installed:**
   - Verify that the necessary libraries, including `propsys.lib`, are available on the build machine for the target architecture (`i686-pc-windows-gnu`).
   - In a cross-compilation scenario, you might need to provide the path to the Windows libraries explicitly. Ensure that the required libraries are accessible in the build environment.

2. **Check Cross Compilation Configuration:**
   - Ensure that your Cargo.toml file and build.rs (if applicable) are correctly configured for cross-compilation to Windows x86.
   - Check if any dependencies or build scripts assume a specific library path or include directory for Windows libraries.

3. **Update Cross Compiler:**
   - Make sure you are using the latest version of the cross-compiler. You can update it using the following command:
     ```bash
     cargo install cross --force
     ```
   - This ensures that you have the latest version, which may include bug fixes and improvements.

4. **Linking with the Correct Library:**
   - If the library name is different or if it requires a specific path, you may need to update your build script or Cargo.toml to reflect the correct library name or path.
   - Check if there are any specific instructions or recommendations for cross-compiling to Windows x86 in the documentation of the libraries you are using.

5. **Search for the Library in the System:**
   - Run a search on the build machine to check if the `propsys.lib` library is present. You can use tools like `find` or `locate` on Linux or the Windows File Explorer search on Windows.

6. **Explicitly Set Library Paths:**
   - In your build.rs or Cargo.toml, you can try explicitly setting the library search path using the `rustc-link-search` attribute. For example:
     ```rust
     println!("cargo:rustc-link-search=native=/path/to/your/libs");
     ```

7. **Check Cross Compilation Target:**
   - Ensure that you are targeting the correct architecture and platform. The target triple for Windows x86 is `i686-pc-windows-msvc` or `i686-pc-windows-gnu`. Confirm that your cross-compilation target matches the target platform.

8. **Check Dependency Documentation:**
   - Review the documentation of the dependencies you are using, especially if they include any platform-specific instructions. The library that requires `propsys.lib` may have specific guidance for cross-compilation.

If after trying these steps the issue persists, consider reaching out to the community of the libraries you are using or checking for any open issues related to cross-compilation in the respective GitHub repositories. Additionally, the Rust forums or the Rust subreddit may provide helpful insights from the community.

**Comment by ysmoradi** [comment]
And this answer from google bard:

The error `cannot find -lpropsys` indicates that the linker cannot find the `propsys` library, which is required for some cross-compilation targets, such as Windows x86.

**Resolution:**

1. Install the `propsys` library on your host system.

   * For Ubuntu:

   ```bash
   sudo apt-get install libpropsys-dev

2. Cross-compile your Rust code.
```
rustup target add x86_64-pc-windows-msvc
cargo build --target=x86_64-pc-windows-msvc
```

3.Run the compiled executable on a Windows x86 machine

**Comment by ysmoradi** [comment]
![image](https://github.com/velopack/velopack/assets/5352980/25ccf4a4-4a1c-4889-af50-5155c2c97f5b)
With this change, error will disappear, but I don't know anything about its side effects.

**Comment by caesay** [comment]
I appreciate the enthusiasm! 

Released in [0.0.208-gd4b199d](https://www.nuget.org/packages/Velopack/0.0.208-gd4b199d).
